src/snowflake/snowpark/mock/plan.py

```diff
@@ -283,23 +283,37 @@ def execute_mock_plan(
         for i in range(1, len(source_plan.set_operands)):
             operand = source_plan.set_operands[i]
             operator = operand.operator
-            if operator in (UNION, UNION_ALL):
-                cur_df = execute_mock_plan(
-                    MockExecutionPlan(operand.selectable, source_plan.analyzer.session),
-                    expr_to_alias,
+            cur_df = execute_mock_plan(
+                MockExecutionPlan(operand.selectable, source_plan.analyzer.session),
+                expr_to_alias,
+            )
+            if len(res_df.columns) != len(cur_df.columns):
+                raise SnowparkSQLException(
+                    f"SQL compilation error: invalid number of result columns for set operator input branches, expected {len(res_df.columns)}, got {len(cur_df.columns)} in branch {i + 1}"
                 )
-                if len(res_df.columns) != len(cur_df.columns):
-                    raise SnowparkSQLException(
-                        f"SQL compilation error: invalid number of result columns for set operator input branches, expected {len(res_df.columns)}, got {len(cur_df.columns)} in branch {i + 1}"
-                    )
-                cur_df.columns = res_df.columns
+            cur_df.columns = res_df.columns
+            if operator in (UNION, UNION_ALL):
                 res_df = pd.concat([res_df, cur_df], ignore_index=True)
                 res_df = (
                     res_df.drop_duplicates().reset_index(drop=True)
                     if operator == UNION
                     else res_df
                 )
                 res_df.sf_types = cur_df.sf_types
+            elif operator == EXCEPT:
+                # Dedup all none rows
+                if res_df.isnull().all(axis=1).where(lambda x: x).count() > 1:
+                    res_df = res_df.drop(index=res_df.isnull().all(axis=1).index[1:])
+                # If there are all none rows in cur_df, drop all none rows in res_df
```

Comment:
I don't think I understand this comment. Could you elaborate it?

Reply:
This is because in pandas `NaN == NaN` evaluates to False. So `NOT IS IN` won't drop rows that are all `NaN/None` (all-none-row's) , neither does `drop_duplicates` remove those rows even if there are multiple.